index.test-d.ts

```diff
@@ -51,6 +148,9 @@ expectType<string>(baseWithVoidAndNonVoidPlugins.bar);
 baseWithVoidAndNonVoidPlugins.unknown;
 
 const BaseWithOptionsPlugin = Base.plugin(withOptionsPlugin);
-const baseWithOptionsPlugin = new BaseWithOptionsPlugin();
+const baseWithOptionsPlugin = new BaseWithOptionsPlugin({
+  version: "1.2.3",
+});
 
 expectType<string>(baseWithOptionsPlugin.getFooOption());
+
```

Comment:
See my resolved discussion above: https://github.com/gr2m/javascript-plugin-architecture-with-typescript-definitions/pull/57/files#r670634320

```suggestion

// Test depth limits of `.defaults()` chaining
const BaseLevelFour = BaseLevelThree.defaults({ defaultFour: 4 });

expectType<{
  version: string;
  defaultOne: string;
  defaultTwo: number;
  defaultThree: string[];
  defaultFour: number;
}>({ ...BaseLevelFour.defaultOptions });

const baseLevelFour = new BaseLevelFour();

expectType<{
  version: string;
  defaultOne: string;
  defaultTwo: number;
  defaultThree: string[];
  defaultFour: number;
  // @ts-expect-error - .options from .defaults() is only supported until a depth of 4
}>({ ...baseLevelFour.options });

```